batchflow/models/tf/encoder_decoder.py

```diff
@@ -4,173 +4,288 @@
 
 from .layers import conv_block
 from . import TFModel
-from .resnet import ResNet18
 
 
 class EncoderDecoder(TFModel):
-    """ Encoder-decoder architecture
-
-    **Configuration**
+    """ Encoder-decoder architecture. Allows to combine features of different models,
+    e.g. ResNet and DenseNet, in order to create new ones with just a few lines of code.
 
+    Parameters
+    ----------
     inputs : dict
-        dict with 'images' (see :meth:`~.TFModel._make_inputs`)
+        Dictionary with 'images' (see :meth:`~.TFModel._make_inputs`)
 
     body : dict
         encoder : dict
             base_class : TFModel
-                a model implementing ``make_encoder`` method which returns tensors
-                with encoded representation of the inputs
-            other args
-                parameters for base class ``make_encoder`` method
+                Model implementing ``make_encoder`` method which returns tensors
+                with encoded representation of the inputs.
+
+            num_stages : int
+                Number of downsampling stages.
+
+            downsample : dict
+                Parameters for downsampling (see :func:`~.layers.conv_block`)
+
+            blocks : dict
+                Parameters for pre-processing blocks:
+
+                base : callable
+                    Tensor processing function. Default is :func:`~.layers.conv_block`.
+                other args : dict
+                    Parameters for the base block.
+
+            other args : dict
+                Parameters for ``make_encoder`` method.
 
         embedding : dict
-            :func:`~.layers.conv_block` parameters for the bottom block
+            base : callable
+                Tensor processing function. Default is :func:`~.layers.conv_block`.
+            other args
+                Parameters for the base block.
 
         decoder : dict
             num_stages : int
-                number of upsampling blocks
+                Number of upsampling blocks.
+
             factor : int or list of int
-                if int, the total upsampling factor for all blocks combined.
-                if list, upsampling factors for each block.
-            layout : str
-                upsampling method (see :func:`~.layers.upsample`)
+                If int, the total upsampling factor for all stages combined.
+                If list, upsampling factors for each stage.
+
+            bridges : bool
```

Comment:
While people all over the world call them `skip`, we call them `bridges`. It doesn't seem right, does it?